src/otp-input/otp-input.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useEffect, useRef, useState } from "react";
+import { OtpInputProps } from "./types";
+import {
+    CTAButton,
+    ErrorContainer,
+    InputContainer,
+    InputField,
+    Wrapper,
+} from "./otp-input.styles";
+
+export const OtpInput = ({
+    cooldownDuration,
+    actionButtonProps,
+    errorMessage,
+    numOfInput,
+    onChange,
+    ...otherProps
+}: OtpInputProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const {
+        disabled,
+        onClick,
+        styleType = "secondary",
+        ...otherCtaProps
+    } = actionButtonProps ?? {};
+    const inputRefs = useRef<(HTMLInputElement | null)[]>([]);
+
+    const [otpValues, setOtpValues] = useState<string[]>(
+        new Array(numOfInput).fill("")
+    );
+    const [countDown, setCountDown] = useState<number>(cooldownDuration);
+    const [enableCountDownTimer, setEnableCountDownTimer] =
+        useState<boolean>(true);
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useEffect(() => {
+        inputRefs?.current[0]?.focus();
+        document.addEventListener("paste", handlePaste);
+        return () => document.removeEventListener("paste", handlePaste);
+    }, []);
+
+    useEffect(() => {
+        if (!enableCountDownTimer || disabled) {
+            return;
+        }
+        let timer = cooldownDuration;
+        setCountDown(timer);
+        const interval = setInterval(() => {
+            timer--;
+            setCountDown(timer);
+            if (timer === 0) {
+                clearInterval(interval);
+                setEnableCountDownTimer(false);
+            }
+        }, 1000);
+        return () => clearInterval(interval);
+    }, [enableCountDownTimer]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleKeyDown = (
```

Comment:
Considering using function curry

```suggestion
    const handleKeyDown = (index: number) => 
            (event: React.KeyboardEvent<HTMLInputElement>) => {
                     // implementations here
            };
```